src/webcouturier/dropdownmenu/vocabularies.py

```diff
@@ -1,31 +1,45 @@
 # -*- coding: utf-8 -*-
 from zope.schema.vocabulary import SimpleTerm, SimpleVocabulary
 from Products.CMFCore.utils import getToolByName
+from plone import api
+#from zope.schema.interfaces import IVocabularyFactory
 
 from webcouturier.dropdownmenu import msg_fact as _
 
-from zope.component.hooks import getSite
-
+try:
+    from zope.app.component.hooks import getSite
+except ImportError:
+    from zope.component.hooks import getSite
```

Comment:
Shouldn't we first try to import from `zope.component.hooks`, and only do the bbb thing upon exception ?